src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -120,9 +120,11 @@ namespace winrt::TerminalApp::implementation
     void TerminalPage::_HandleOpenSettings(const IInspectable& /*sender*/,
                                            const TerminalApp::ActionEventArgs& args)
     {
-        // TODO:GH#2557 Add an optional arg for opening the defaults here
-        _LaunchSettings(false);
-        args.Handled(true);
+        if (const auto& realArgs = args.ActionArgs().try_as<TerminalApp::OpenSettingsArgs>())

```

Comment:
@zadjii-msft I ran into a weird problem here and that's why it's a draft.

When I set a breakpoint here, it does get hit, but the if-statement fails. I can't actually figure out why. Any thoughts on where to look or what may be missing?

Reply:
You need to make sure to add support to `AppKeyBindingsSerialization` to parse `openSettings` as an `ActionAndArgs`. Right now it's just got an empty (or null) `ActionAndArgs`, so it can't `try_as` successfully 😉